### ISSUE TITLE
Crackling/Static on microphone when lighting earpods plugged in after mic init

### ISSUE THREAD

**Issue by Meekohi** [AudioKit/AudioKit#1213]
When using wired lightning earpods, if they are plugged into the device after the microphone is initialized, mic recordings contains some popping/crackling/static sounds (seems amplitude dependent -- worse the louder the recording).

Easiest way to test this is using the playground "Record Demo" app, but it is much easier to hear if you disconnect the Moog filter:
```
## ViewController.swift
- mainMixer = AKMixer(moogLadder, micBooster)
+ mainMixer = AKMixer(player, micBooster)
```

Curious if anyone has an idea what is going on here. Searching for work-arounds in the meantime.

**Comment by aure** [comment]
Can you provide a sample project?

**Comment by Meekohi** [comment]
Hey @aure what would be a good way to package it up for you? What I've been doing is:
```
1. Download https://github.com/AudioKit/AudioKit/releases/download/v4.0.4/AudioKit-iOS-4.0.4.zip
2. Open Examples/Recorder/Recorder.xcodeproj
3. Change L81 of ViewController.swift from
  mainMixer = AKMixer(moogLadder, micBooster)
to
  mainMixer = AKMixer(player, micBooster)
4. Run on a physical iOS device.
  4a. If you have the headphones plugged in when the app starts, it sounds good
  4b. If you wait and plug the headphones in after the app already launched, there is static/crackling in the audio.
```

**Comment by Meekohi** [comment]
Also to clarify, I've only seen this behavior on devices with lightning-plug earpods. Had some other devs test and they say it is worst on iPhone X, I'm testing on an iPhone 8.

**Comment by marcosgriselli** [comment]
@Meekohi is it only happening when you do the earpods thing? I've changed that one line and have cracking/static sound on my recording using nothing but the phone 😕.

**Comment by Meekohi** [comment]
Have to admit now I'm a bit lost.

I just got some headphones to test with and confirmed I see this problem with them. Then I tested the app normally (no headphones, just standard iPhone 8) and was surprised to still hear the problem. Perhaps I didn't test this very carefully originally, or I have updated the iOS version in the meantime which introduced this problem?

**Comment by marcosgriselli** [comment]
The same issue happens to me when I use the exact same Recorder demo but bypass the `AKMoogLadder` filter doing `moogLadder.stop()` after the setup. 

This issue doesn't happen on iPad. I tested on one iPhone X and iPhone 6s with the same result, that static/cracking sound. But I tested the exact same code on two different iPads Pros and it worked correctly, perfect recording playback.

**Comment by dave234** [comment]
My gues is that it’s related to the hard coded 44100 sample rate issue.

**Comment by marcosgriselli** [comment]
@dave234 That was my guess as well, but I tried `AKSettings.sampleRate = 48000` and it didn't solve it.

**Comment by dave234** [comment]
I don’t think AKSettings.sampleRate is respected throughout the project.

**Comment by marcosgriselli** [comment]
@dave234 I see, is there any way I can tweak this value or is it part of the AudioKit core functionality? I'm trying to find a solution but I could use some guidance.

**Comment by mscheffel** [comment]
Had the same problem. Our fix was to set AudioKit.defaultToSpeaker to false everytime we connected the microphone. AudioKit has to be stopped to set this value!!! This is important - otherwise you will have crackling audio all the time.
Maybe this could be documented at defaultToSpeaker, to make sure developers understand the risk of crackling audio. Additionally we tested this with many devices (iPhone 5s -> iPhone X). The behaviour only appears to happen on devices newer than iPhone 6 (6s and newer).
